File: FinalProject/app/src/main/java/com/example/finalproject/fragments/ItemCharacteristicsFragment.java
package com.example.finalproject.fragments;

import android.graphics.Bitmap;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.finalproject.items.Item;
import com.example.finalproject.R;

import org.jetbrains.annotations.NotNull;

public class ItemCharacteristicsFragment extends Fragment {
    private Item item;
    private boolean mode;

    public ItemCharacteristicsFragment(Item item){
        this.item=item;
        mode=true;
    }

    public ItemCharacteristicsFragment(Item item, boolean mode){
        this.item=item;
        this.mode=mode;
    }

    @Nullable
    @org.jetbrains.annotations.Nullable
    @Override
    public View onCreateView(@NonNull @NotNull LayoutInflater inflater, @Nullable @org.jetbrains.annotations.Nullable ViewGroup container, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_item_characteristics, container, false);
    }

    @Override
    public void onViewCreated(@NonNull @org.jetbrains.annotations.NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ImageView img=getView().findViewById(R.id.category_image);
        img.setImageBitmap(Bitmap.createScaledBitmap(MainActivity.textures[6][item.getCategory()],
                MainActivity.categoryImageWidth, MainActivity.categoryImageWidth, false));
        TextView name = (TextView) getView().findViewById(R.id.name_field);
        TextView category = (TextView) getView().findViewById(R.id.category_field);
        TextView cost = (TextView) getView().findViewById(R.id.cost_field);
        name.setText(item.getName()+"");
        category.setText(MainActivity.categories.get(item.getCategory())+"");
        if (mode)
            cost.setText(item.getCostSell()+"");
        else cost.setText(item.getCostBuy()+"");
    }

    public Item getItem() {
        return item;
    }

    public void setItem(Item item) {
        this.item = item;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/fragments/ItemCharacteristicsFragment.kt
package com.example.finalproject.fragments

import android.graphics.Bitmap
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.finalproject.R
import com.example.finalproject.items.Item
import kotlin.math.cos

class ItemCharacteristicsFragment(val item: Item, private val mode: Boolean = true) : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_item_characteristics, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val img: ImageView = requireView().findViewById(R.id.category_image)
        img.setImageBitmap(
            Bitmap.createScaledBitmap(
                MainActivity.textures[6][item.category],
                MainActivity.categoryImageWidth, MainActivity.categoryImageWidth, false
            )
        )
        val name: TextView = requireView().findViewById(R.id.name_field)
        val category: TextView = requireView().findViewById(R.id.category_field)
        val cost: TextView = requireView().findViewById(R.id.cost_field)
        name.text = item.name
        category.text = item.category.toString()
        if (mode)
            cost.text = item.costSell.toString()
        else
            cost.text = item.costBuy.toString()
    }
}